source/simulator.hpp

```diff
@@ -13,10 +13,10 @@ class Simulator {
 public:
     Simulator();
     ~Simulator();
-    Device* add_device(std::string a_name, DeviceType a_type);
-    void add_flow(Device* a_from, Device* a_to);
-    void add_link(Device* a_from, Device* a_to, std::uint32_t a_speed_mbps,
-                  std::uint32_t a_delay);
+    IRoutingDevice* add_device(std::string a_name, DeviceType a_type);
+    void add_flow(IRoutingDevice* a_from, IRoutingDevice* a_to);
```

Comment:
And the same thing here:
```suggestion
    void add_flow(ISender* a_from, IReceiver* a_to);
```

Reply:
Yes, my miss